pycode/memilio-generation/memilio/generation/scanner.py

```diff
@@ -0,0 +1,322 @@
+        Information: age_group
+
+        @param node Current node represented as a Cursor object.
+        @param intermed_repr Dataclass used for saving the extracted model features.
+        """
+        for base in node.get_children():
+            if base.kind != CursorKind.CXX_BASE_SPECIFIER:
+                continue
+            for base_template_arg in base.get_children():
+                if base_template_arg.kind == CursorKind.TYPE_REF and "AgeGroup" in base_template_arg.spelling:
+                    for child in base_template_arg.get_definition().get_children():
+                        if child.kind == CursorKind.CXX_BASE_SPECIFIER:
+                            intermed_repr.age_group["base"] = child.get_definition(
+                            ).type.spelling
+                        elif child.kind == CursorKind.CONSTRUCTOR:
+                            intermed_repr.age_group["init"] = [
+                                arg.spelling
+                                for arg in child.type.argument_types()]
+
+    def check_constructor(
+        self: Self, node: Cursor,
+            intermed_repr: IntermediateRepresentation) -> None:
+        """
+        Inspect the nodes of kind CONSTRUCTOR and writes needed information into intermed_repr.
+        Information: intermed_repr.init
+
+        @param node Current node represented as a Cursor object.
+        @param intermed_repr Dataclass used for saving the extracted model features.
+        """
+        if node.spelling == intermed_repr.model_class:
+            init = {"type": [], "name": []}
+            for arg in node.get_arguments():
+                tokens = []
+                for token in arg.get_tokens():
+                    tokens.append(token.spelling)
+                init["type"].append(" ".join(tokens[:-1]))
+                init["name"].append(tokens[-1])
+            intermed_repr.model_init.append(init)
+
+    def check_type_alias(
+        self: Self, node: Cursor,
+            intermed_repr: IntermediateRepresentation) -> None:
+        """
+        Inspect the nodes of kind TYPE_ALIAS_DECL and writes needed information into intermed_repr.
+        Information: intermed_repr.parameterset
+
+        @param node Current node represented as a Cursor object.
+        @param Dataclass used for saving the extracted model features.
+        """
+        if node.spelling == self.config.parameterset:
+            intermed_repr.parameterset = node.spelling
+
+    def check_struct(
+        self: Self, node: Cursor,
+            intermed_repr: IntermediateRepresentation) -> None:
+        """ Not used yet."""
+        pass
+
+    def finalize(self: Self, intermed_repr: IntermediateRepresentation) -> None:
+        """
+        Finalize the IntermediateRepresenation as last step of the Scanner.
+        Writes needed information from config into intermed_repr, delets unnecesary enums and ckecks for missing model features.
```

Comment:
Please be consistent on infinitive and third person (writes). I think generally we use third person. However it may not be really consistent in our documentation everywhere at the moment.